### ISSUE TITLE
functional.fbeta_score memory use

### ISSUE THREAD

**Issue by rpgit12** [PyTorchLightning/metrics#1053]
### To Reproduce

Steps to reproduce the behavior...

### Environment

- TorchMetrics version (and how you installed TM, e.g. `conda`, `pip`, build from source):
- Python & PyTorch Version (e.g., 1.0):
- Any other relevant information such as OS (e.g., Linux):

**Comment by Borda** [comment]
@rpgit12 do you have any question?

**Comment by rpgit12** [comment]
Sorry for inadvertently hitting submit before entering text ...

I'm calculating macro f2 score as per the below. I believe this should be fairly straightforward to calculate, but the torchmetrics implementation usages excessive memory (attempting to allocate 32 GBs.) This is intended for an macro level f2 score, multilabel (not multiclass.)

Can you help confirm this is indeed a problem with the fbeta_score() implementation and I'm using the API correctly?

```
target = torch.randint(2, (1000, 2000))
preds = torch.randint(2, target.shape)
tm.functional.fbeta_score(preds, target, beta=2, num_classes=target.shape[1], multiclass=False, average='macro')

```

**Comment by SkafteNicki** [comment]
Hi @rpgit12,
You are completely correct that `fbeta_score` currently are using large amount of memory when the input contains many classes/labels. The problem is the tensor created here:
https://github.com/PyTorchLightning/metrics/blob/0495862cae1fd4caa9b77e97bbb0a9d783ecd881/torchmetrics/utilities/data.py#L105-L111
which in your case will be [1000, 2000, 2000] tensor that fills around 30 GB as you report. We convert to one-hot format to be able to easy calculate top-k scores if the user requests it.

The bad news is that there is nothing to do about it right now. The good news is that we have planned a major refactor of the classification package #1001 (which will start next week) where we definitely should address this shortcoming.

Thanks for reporting this :]

**Comment by rpgit12** [comment]
@SkafteNicki - thank you for confirming.  I think there is still another issue, as the F2 score returned between individual labels and the macros score doesn't seem to match (what I'd expect, at least.)

Below an example; I would expect that the arithmetic mean of the individual scores matches the macro f2 score for all labels combined.

```
from torchmetrics.functional import fbeta_score

target = torch.randint(2, (1000, 3))
preds = torch.randint(2, target.shape)
f2_individual = [ fbeta_score(preds[:, _], target[:, _], beta=2, num_classes=target.shape[1], average='macro').item() for _ in range(target.shape[1])]
f2_combined   = fbeta_score(preds, target, beta=2, num_classes=target.shape[1], multiclass=False, average='macro')

print("f2, per label       ", np.mean(f2_individual), f2_individual)
print("f2, combined        ", f2_combined.item())
```

Thanks again for looking at this!

**Comment by SkafteNicki** [comment]
This is another inconsistency that I hope to address in the upcoming refactor. Basically the problem is that the individual is calculates as an average of the positive and negative class, where as it should only be calculated over the positive class. A slightly adjusted version of your code does this correctly:
```python
import torch
from torchmetrics.functional import fbeta_score
import numpy as np

target = torch.randint(2, (1000, 3))
preds = torch.randint(2, target.shape)
f2_individual = [fbeta_score(preds[:, _], target[:, _], beta=2, average=None, num_classes=2)[-1].item() for _ in range(target.shape[1])]
f2_combined   = fbeta_score(preds, target, beta=2, num_classes=target.shape[1], multiclass=False, average='macro')

print("f2, per label       ", np.mean(f2_individual), f2_individual)
print("f2, combined        ", f2_combined.item())
# f2, per label        0.5380443930625916 [0.5453456044197083, 0.5286701917648315, 0.5401173830032349]
# f2, combined         0.5380444526672363
```
Again sorry for the troubles this is giving you.
